### PR TITLE
Cast numpy's number types to Python

### LINKED ISSUE #14186: mypy 1.13.0 fails in `branch-3.7`
Fails with:
```py
mypy 1.13.0 (compiled: yes)
src/bokeh/palettes.py: note: In function "interp_palette":
src/bokeh/palettes.py:1676:23: error: Argument 1 to "RGB" has incompatible type
"*tuple[unsignedinteger[_8Bit], unsignedinteger[_8Bit], unsignedinteger[_8Bit], floating[Any]]";
expected "int"  [arg-type]
        return tuple(RGB(*args).to_hex() for args in zip(r, g, b, a))
                          ^~~~
src/bokeh/palettes.py:1676:23: error: Argument 1 to "RGB" has incompatible type
"*tuple[unsignedinteger[_8Bit], unsignedinteger[_8Bit], unsignedinteger[_8Bit], floating[Any]]";
expected "float"  [arg-type]
        return tuple(RGB(*args).to_hex() for args in zip(r, g, b, a))
                          ^~~~
Found 2 errors in 1 file (checked 656 source files)
```

### PR DESCRIPTION
**Pull request by mattpap**
fixes #14186

### CODE REVIEW

File: src/bokeh/colors/color.py [hunk 1]
```diff
@@ -207,7 +208,12 @@ class RGB(Color):
 
     '''
 
-    def __init__(self, r: int, g: int, b: int, a: float = 1.0) -> None:
+    r: int
+    g: int
+    b: int
+    a: float
+
+    def __init__(self, r: int | np.uint8, g: int | np.uint8, b: int | np.uint8, a: float | np.floating = 1.0) -> None:
```
Comment (hoxbro) [inline src/bokeh/colors/color.py]:
```suggestion
    def __init__(self, r: int | np.integer, g: int | np.integer, b: int | np.integer, a: float | np.floating = 1.0) -> None:
```

I don't think it should be limited to `np.uint8` even though that is the correct range for RGB.

**Review by hoxbro — COMMENTED** [review]
This is caused by Numpy 2.2 release, correct?

**Comment by mattpap** [comment]
> This is caused by Numpy 2.2 release, correct?

I'm not sure. Either that or mypy 1.13.

**Comment by hoxbro** [comment]
I think it is related to the discussion in https://github.com/numpy/numpy/issues/27957.

**Review by hoxbro — APPROVED** [review]

**Pull request merged**